dmoj/problem.py

```diff
@@ -330,17 +346,33 @@ def _run_generator(self, gen, args=None):
         parse_helper_file_error(proc, executor, 'generator', stderr, time_limit, memory_limit)
 
     def input_data(self):
+        return self.input_data_fd().to_bytes()
+
+    def input_data_fd(self):
+        if not self._input_data_fd:
+            self._input_data_fd = self._make_input_data_fd()
+        return self._input_data_fd
+
+    def _make_input_data_fd(self):
```

Comment:
Maybe this needs to be typed to return `MemoryIO` now?

Reply:
Any file object will do, `MemoryIO` is just a special hack.